SConstruct

```diff
@@ -641,6 +641,9 @@ if selected_platform in platform_list:
             }
         )
 
+    # enable test framework globally and inform it of configuration method
+    env.Append(CPPDEFINES=["DOCTEST_CONFIG_IMPLEMENT"])
```

Comment:
Can we have this in `main/tests/SCsub`? Not sure why it would have to be in `SConstruct`.

Reply:
It needs to be global for the project so that other parts of the engine can use the tests